dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/enrollment/DefaultEnrollmentService.java

```diff
@@ -162,9 +173,26 @@ public Page<Enrollment> findEnrollments(
             params.getFields(),
             params.isIncludeDeleted(),
             queryParams.getOrganisationUnitMode());
+
+    if (queryParams.getTrackedEntity() != null) {
+      addTrackedEntityAudit(queryParams.getTrackedEntity(), enrollments);
```

Comment:
```suggestion
    if (queryParams.getTrackedEntity() != null && !enrollments.isEmpty()) {
      trackedEntityAuditService.addTrackedEntityAudit(
                READ, getCurrentUserDetails().getUsername(), queryParams.getTrackedEntity()))
```

In my opinion here we are checking too many things.
If we arrived at this point it means that `trackedEntity` was declared in the params so all the enrollments in the result must be of that `trackedEntity`.
If we make sure the query returned any result, then we can add the audit for the tracked entity.